src/java.compiler/share/classes/javax/lang/model/element/PackageElement.java

```diff
@@ -32,6 +32,28 @@
  * Represents a package program element.  Provides access to information
  * about the package and its members.
  *
+ * @apiNote
+ * The represented package may have an explicit {@linkplain
+ * javax.lang.model.util.Elements#getFileObjectOf(Element) reference
+ * representation} (either source code or executable output) or may be
+ * created from implicit information. The explicit and standalone
+ * source code construct for a package is typically a {@code
+ * package-info.java} file (JLS {@jls 7.4.1}). Implicit information is
+ * used to model {@linkplain #isUnnamed unnamed packages}.
```

Comment:
you mention unnamed packages but not named packages without `package-info.java`

Reply:
Added discussion on that point; will reflow paragraphs later.